jgrapht-core/src/main/java/org/jgrapht/alg/lca/BinaryLiftingLCAFinder.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.lca;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.LCAAlgorithm;
+
+import java.util.*;
+
+import static org.jgrapht.util.MathUtil.log2;
+
+/**
+ * Algorithm for computing lowest common ancestors in rooted trees and forests using the binary lifting method.
+ *
+ * Preprocessing Time complexity: $O(|V| log(|V|))
+ * Preprocessing Memory complexity:  $O(|V| log(|V|))
```

Comment:
You can have memory requirements, etc. but complexity is referred as "space complexity" in academic papers.